packages/back-end/src/events/handlers/webhooks/EventWebHookNotifier.ts

```diff
@@ -0,0 +1,271 @@
+   * @param payload
+   * @param eventWebHook
+   */
+  private static async sendDataToWebHook<DataType>({
+    payload,
+    eventWebHook,
+  }: {
+    payload: DataType;
+    eventWebHook: EventWebHookInterface;
+  }): Promise<EventWebHookResult> {
+    try {
+      const { url, signingKey } = eventWebHook;
+
+      const signature = getEventWebHookSignatureForPayload({
+        signingKey,
+        payload,
+      });
+
+      const res = await fetch(url, {
+        headers: {
+          "Content-Type": "application/json",
+          "X-GrowthBook-Signature": signature,
+        },
+        method: "POST",
+        body: JSON.stringify(payload),
+      });
+
+      if (!res.ok) {
+        // Server error
+        return {
+          result: "error",
+          statusCode: res.status,
+          error: res.statusText,
+        };
+      }
+
+      // Success
+      return {
+        result: "success",
+        statusCode: res.status,
+      };
+    } catch (e) {
+      // Unknown error
+      console.error("Unknown Error", e);
+
+      return {
+        result: "error",
+        statusCode: null,
+        error: e.message,
+      };
+    }
+  }
+
+  // region Result handling
+
+  private static async handleWebHookSuccess(
+    job: Job<EventWebHookJobData>,
+    successResult: EventWebHookSuccessResult,
+    organizationId: string,
+    payload: Record<string, unknown>
+  ): Promise<void> {
+    const { eventWebHookId } = job.attrs.data;
+
+    await updateEventWebHookStatus(eventWebHookId, {
+      state: "success",
+    });
+
+    await createEventWebHookLog({
+      eventWebHookId,
+      organizationId,
+      payload,
+      result: {
+        state: "success",
+        responseCode: successResult.statusCode,
+      },
+    });
+  }
+
+  private static async handleWebHookError(
+    job: Job<EventWebHookJobData>,
+    errorResult: EventWebHookErrorResult,
+    organizationId: string,
+    payload: Record<string, unknown>
+  ): Promise<void> {
+    const { eventWebHookId } = job.attrs.data;
+
+    await updateEventWebHookStatus(eventWebHookId, {
+      state: "error",
+      error: errorResult.error,
+    });
+
+    await createEventWebHookLog({
+      eventWebHookId,
+      organizationId,
+      payload,
+      result: {
+        state: "error",
+        error: errorResult.error,
+        responseCode: errorResult.statusCode,
+      },
+    });
+
+    await EventWebHookNotifier.retryJob(job);
+  }
+
+  /**
+   * Retries the job. Should only be called when a job has failed.
+   * Retries up to 3 times.
+   * Retries are as follows:
+   *  1. 30 seconds later
+   *  2. 5 minutes later
+   *  3. 5 minutes later
+   * @param job
+   * @private
+   */
+  private static async retryJob(job: Job<EventWebHookJobData>) {
+    if (job.attrs.data.retryCount >= 3) {
+      // If it failed 3 times, give up
+      return;
+    }
+
+    job.attrs.data.retryCount++;
+
+    let nextRunAt = Date.now();
+    if (job.attrs.data.retryCount === 0) {
```

Comment:
Yes. I had a similar question when looking at the existing logic, but after looking where `retryCount` is defined, you'll notice that the job is defined with arguments and when a job is created we include the `retryCount`. https://github.com/growthbook/growthbook/pull/760/files/ca63e0da49a272997cca4c3f4fec5de319da2685#diff-bad7c46f3aeff520109b50057a40bd6e330d448da83caa8a95668ac870c85b92R34

If you wanted to be sure, you can pull down the code and test with the instructions and add some logging. To more quickly test, it would make sense to not increment `nextRunAt` so it's faster, but you should be able to track the retry count.

Reply:
I think you misunderstood my comment. I was referring to the line of code at line 255 - we're incrementing the count before checking that it's equal to `0`

However I did pull down the branch and try to run it locally with appropriate logging since that would be a great exercise for me. It did confirm my findings:

<img width="524" alt="Screen Shot 2022-11-30 at 10 29 12 PM" src="https://user-images.githubusercontent.com/2374625/204959500-f4fe39ae-aa5f-4bc7-a538-137ee53e6193.png">

Reply:
Thanks for testing it out, I appreciate it. I think you're right here, I had "cleaned up" this section and I think I introduced a bug, great eye. Let me take another look. 👀

Reply:
I pushed up a change, moving the `retryCount` incrementing down to where we set the `nextRunAt` (and after we check its value) and it looks like it should be ok now.